packages/contracts/contracts/voting/majority/IMajorityVoting.sol

```diff
@@ -43,37 +43,93 @@ import "../../core/IDAO.sol";
 ///  For minimal values, $\ge$ comparison is used. This **does** include the minimum participation value. E.g., for $\texttt{minParticipation} = 40\%$ and $N_\text{total} = 10$, the criterion is fulfilled if 4 out of 10 votes were casted.
 /// @dev This contract implements the `IMajorityVoting` interface.
 interface IMajorityVoting {
+    /// @notice Vote options that a voter can chose from.
+    /// @param None The default option state of a voter indicating the absence of from the vote. This option neither influences support nor participation.
+    /// @param Abstain This option does not influence the support but counts towards participation.
+    /// @param Yes This option increases the support and counts towards participation.
+    /// @param No This option decreases the support and counts towards participation.
     enum VoteOption {
         None,
         Abstain,
         Yes,
         No
     }
 
+    /// @notice The vote mode in which the plugin can be configured.
+    /// @param Standard In standard mode, early execution and vote replacement are disabled.
+    /// @param EarlyExecution In early execution mode, a proposal can be executed early if the vote outcome cannot mathematically change by more voters voting.
+    /// @param VoteReplacment In vote replacement mode, voters can change their vote multiple times and only the latest vote option is tallied.
+    enum VoteMode {
+        Standard,
+        EarlyExecution,
+        VoteReplacement
+    }
+
+    /// @notice A container for the plugin-wide proposal vote settings.
+    /// @param voteMode A parameter to select the vote mode.
+    /// @param supportThreshold The support threshold value.
+    /// @param minParticipation The minimum participation value.
+    /// @param minDuration The minimum duration of the proposal vote in seconds.
+    /// @param minProposerVotingPower The minimum voting power required to create a proposal.
+    struct PluginSettings {
+        VoteMode voteMode;
+        uint64 supportThreshold;
+        uint64 minParticipation;
+        uint64 minDuration;
+        uint256 minProposerVotingPower;
+    }
+
+    /// @notice A container for proposal-related information.
+    /// @param executed Wheter the proposal is executed or not.
+    /// @param configuration The proposal-specific vote settings at the time of the proposal creation.
+    /// @param tally The vote tally of the proposal.
+    /// @param voters The votes casted by the voters.
+    /// @param actions The actions to be executed when the proposal passes.
     struct Proposal {
         bool executed;
+        Configuration configuration;
```

Comment:
What's the difference between configuration and settings? This sounds very ambiguous. 

Maybe `ProposalParameters parameters` as opposed to the global settings from the plugin itself?

Reply:
Amended in b7630a2649ec40a31b8f23b0a416e42754335d17

Reply:
I don't like this idea. at first it seems good, but when I look at the following code michael just did:

```
(proposal_.parameters.startDate
```

It's confusing. `proposal` is a storage state variable. having it `parameters ` doesn't make sense as parameters correspond to function parameters.  very confusing.

Reply:
`VotingSettings` will be applied as `ProposalParameters` of a proposal vote during proposal creation.
I think this is the clearest way of naming them and I will explain it in the docs.